--- old/nsm_sensor_ps-restart	2014-05-01 19:06:56.014349810 +0000
+++ new/nsm_sensor_ps-restart	2014-05-11 20:30:17.381365314 +0000
@@ -367,6 +367,7 @@
 	HTTP_AGENT_ENABLED="yes"
 	PCAP_SIZE="150MiB"
 	PCAP_RING_SIZE="64MiB"
+	PCAP_SNAP_LENGTH="65535"
 
 	# add parameters to $SENSOR_CONF if they don't already exist
 	grep PCAP_ENABLED $SENSOR_CONF >/dev/null 		|| echo "PCAP_ENABLED=\"yes\""		>> $SENSOR_CONF
@@ -381,6 +382,7 @@
 	grep HTTP_AGENT_ENABLED $SENSOR_CONF >/dev/null 	|| echo "HTTP_AGENT_ENABLED=\"yes\"" 	>> $SENSOR_CONF
 	grep PCAP_SIZE $SENSOR_CONF >/dev/null			|| echo "PCAP_SIZE=\"150MiB\""		>> $SENSOR_CONF
 	grep PCAP_RING_SIZE $SENSOR_CONF >/dev/null		|| echo "PCAP_RING_SIZE=\"64MiB\""	>> $SENSOR_CONF				
+	grep PCAP_SNAP_LENGTH $SENSOR_CONF >/dev/null           || echo "PCAP_SNAP_LENGTH=\"65535\""    >> $SENSOR_CONF
 
 	# load in sensor specific details
 	. $SENSOR_CONF
@@ -399,15 +401,11 @@
 			chown $SENSOR_USER:$SENSOR_GROUP $SENSOR_LOG_DIR/dailylogs/$TODAY
 			chmod 775 $SENSOR_LOG_DIR/dailylogs/$TODAY
 		fi
-		# Did the user supply a BPF?
-	        if [ -s /etc/nsm/$SENSOR/bpf-pcap.conf ]; then
-	                # User supplied a BPF, so we need to convert it
+	        # Convert BPF
 	                BPF=`grep -v "^#" /etc/nsm/$SENSOR/bpf-pcap.conf`
-	                tcpdump -i $SENSOR_INTERFACE_SHORT -dd $BPF > /etc/nsm/$SENSOR/bpf-pcap.ops 2>/dev/null
+	        tcpdump -i $SENSOR_INTERFACE_SHORT -dd $BPF -s $PCAP_SNAP_LENGTH > /etc/nsm/$SENSOR/bpf-pcap.ops 2>/dev/null
 	                BPF_OPTION="--filter /etc/nsm/$SENSOR/bpf-pcap.ops"
-	        else
-	                BPF_OPTION=""
-	        fi
+
 		$ACTION_PCAP "netsniff-ng" "-i $SENSOR_INTERFACE_SHORT -o $SENSOR_LOG_DIR/dailylogs/$TODAY/ --user `id -u $SENSOR_USER` --group `id -g $SENSOR_GROUP` -s --prefix snort.log. --verbose --ring-size $PCAP_RING_SIZE --interval $PCAP_SIZE $PCAP_OPTIONS $BPF_OPTION" "$PROCESS_PID_DIR/$SENSOR/netsniff-ng.pid" "$PROCESS_LOG_DIR/$SENSOR/netsniff-ng.log" "netsniff-ng (full packet data)"
 	fi
 
