--- old/sosetup	2014-05-01 19:06:37.534197117 +0000
+++ new/sosetup	2014-05-01 19:11:09.724447208 +0000
@@ -104,6 +104,7 @@
 # PCAP_RING_SIZE is the RX_RING size for buffering packets
 # Default is 64MB
 PCAP_RING_SIZE="64"
+PCAP_SNAP_LENGTH="65535"
 # WARN_DISK_USAGE is the percentage of disk usage at which the NSM scripts warn the disk is getting full
 WARN_DISK_USAGE=80
 # CRIT_DISK_USAGE is the percentage of disk usage at which the NSM scripts begin purging old files
@@ -1166,6 +1167,7 @@
 	echo "PCAP_OPTIONS=\"$PCAP_OPTIONS\"" 			>> /etc/nsm/"$SENSORNAME"/sensor.conf
 	echo "PCAP_SIZE=${PCAP_SIZE}MiB" 			>> /etc/nsm/"$SENSORNAME"/sensor.conf
 	echo "PCAP_RING_SIZE=${PCAP_RING_SIZE}MiB"		>> /etc/nsm/"$SENSORNAME"/sensor.conf
+	echo "PCAP_SNAP_LENGTH=${PCAP_SNAP_LENGTH}MiB"		>> /etc/nsm/"$SENSORNAME"/sensor.conf
 	echo "IDS_LB_PROCS=$IDS_LB_PROCS" 			>> /etc/nsm/"$SENSORNAME"/sensor.conf
 	echo "PCAP_ENABLED=\"$PCAP_ENABLED\""			>> /etc/nsm/"$SENSORNAME"/sensor.conf
 	echo "PCAP_AGENT_ENABLED=\"$PCAP_AGENT_ENABLED\""	>> /etc/nsm/"$SENSORNAME"/sensor.conf
